task2/rest-api-basics/web/src/main/java/com/epam/esm/web/controller/GiftCertificateController.java

```diff
@@ -0,0 +1,101 @@
+package com.epam.esm.web.controller;
+
+import com.epam.esm.dto.GiftCertificateDto;
+import com.epam.esm.service.GiftCertificateService;
+import com.epam.esm.service.exception.ServiceException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@RestController
+public class GiftCertificateController {
+    private static final Logger logger = LogManager.getLogger(GiftCertificateController.class);
+
+    @Autowired
+    private GiftCertificateService giftCertificateService;
+
+    private Map<String, String[]> multiValueMapToMap(MultiValueMap<String, String> requestParams) {
```

Comment:
@igor-zhuravlev Strange construction.

Reply:
It convert request query params to Map<K, List<V>>, because i have the same params like sort (&sort=name,asc&sort=date,desc)

Reply:
> @igor-zhuravlev Strange construction.

It convert request params to Map<String, List<>>, because I have the same params like sort (&sort=name,asc&sort=date,desc)